<launch>
    <!-- Node Settings -->
    <arg name="output" default="screen"/>
    <arg name="imu" default="imu"/>

    <!-- IMU Settings -->
    <arg name="device" default="/dev/ttyACM0" />

    <!-- Frame ID for messages -->
    <arg name="frame_id" default="$(arg imu)"/>

    <!-- Baudrate of serial comms (see manual for allowed values) -->
    <arg name="baudrate" default="921600"/>

    <!-- Decimation value x, where freq = 1000 / x -->
    <arg name="imu_decimation" default="10"/>
    <arg name="filter_decimation" default="5"/>

    <!-- Enable/Disable the filter -->
    <arg name="enable_filter" default="false"/>

    <!-- Enable/Disable filter updates -->
    <arg name="enable_accel_update" default="false"/>
    <arg name="enable_mag_update" default="false"/>

    <node pkg="imu_3dm_gx4" name="$(arg imu)" type="imu_3dm_gx4" output="$(arg output)">
        <param name="device" type="string" value="$(arg device)" />
        <param name="baudrate" type="int" value="$(arg baudrate)" />
        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="imu_decimation" type="int" value="$(arg imu_decimation)" />
        <param name="filter_decimation" type="int" value="$(arg filter_decimation)"/>
        <param name="enable_filter" type="bool" value="$(arg enable_filter)"/>
        <param name="enable_accel_update" type="bool" value="$(arg enable_accel_update)"/>
        <param name="enable_mag_update" type="bool" value="$(arg enable_mag_update)"/>
    </node>
</launch>
